cmd/gctcli/trades.go

```diff
@@ -0,0 +1,756 @@
+		})
+	if err != nil {
+		return err
+	}
+	fmt.Printf("%v\t| Beginning stream retrieving trades in 1 hour batches from %v to %v\n",
+		time.Now().Format(time.Kitchen),
+		s.UTC().Format(common.SimpleTimeFormatWithTimezone),
+		e.UTC().Format(common.SimpleTimeFormatWithTimezone))
+	fmt.Printf("%v\t| If you have provided a large time range, please be patient\n\n",
+		time.Now().Format(time.Kitchen))
+	for {
+		resp, err := result.Recv()
+		if err != nil {
+			return err
+		}
+		if len(resp.Trades) == 0 {
+			break
+		}
+		fmt.Printf("%v\t| Processed %v trades between %v and %v\n",
+			time.Now().Format(time.Kitchen),
+			len(resp.Trades),
+			resp.Trades[0].Timestamp,
+			resp.Trades[len(resp.Trades)-1].Timestamp)
+	}
+
+	fmt.Printf("%v\t| Trade retrieval complete! Process took %v\n",
+		time.Now().Format(time.Kitchen),
+		time.Since(streamStartTime))
+
+	return nil
+}
+
+func convertSavedTradesToCandles(c *cli.Context) error {
+	if c.NArg() == 0 && c.NumFlags() == 0 {
+		return cli.ShowCommandHelp(c, "convertsavedtradestocandles")
+	}
+
+	var exchangeName string
+	if c.IsSet("exchange") {
+		exchangeName = c.String("exchange")
+	} else {
+		exchangeName = c.Args().First()
+	}
+	if !validExchange(exchangeName) {
+		return errInvalidExchange
+	}
+
+	var currencyPair string
+	if c.IsSet("pair") {
+		currencyPair = c.String("pair")
+	} else {
+		currencyPair = c.Args().Get(1)
+	}
+	if !validPair(currencyPair) {
+		return errInvalidPair
+	}
+
+	p, err := currency.NewPairDelimiter(currencyPair, pairDelimiter)
+	if err != nil {
+		return err
+	}
+
+	var assetType string
+	if c.IsSet("asset") {
+		assetType = c.String("asset")
+	} else {
+		assetType = c.Args().Get(2)
+	}
+
+	if !validAsset(assetType) {
+		return errInvalidAsset
+	}
+
+	if c.IsSet("interval") {
+		candleGranularity = c.Int64("interval")
+	} else if c.Args().Get(3) != "" {
+		candleGranularity, err = strconv.ParseInt(c.Args().Get(3), 10, 64)
+		if err != nil {
+			return err
+		}
+	}
+
+	if !c.IsSet("start") {
+		if c.Args().Get(4) != "" {
+			startTime = c.Args().Get(4)
+		}
+	}
+
+	if !c.IsSet("end") {
+		if c.Args().Get(5) != "" {
+			endTime = c.Args().Get(5)
+		}
+	}
+
+	var sync bool
+	if c.IsSet("sync") {
+		sync = c.Bool("sync")
+	}
+
+	var force bool
+	if c.IsSet("force") {
+		force = c.Bool("force")
+	}
+
+	if force && !sync {
+		return errors.New("cannot forcefully overwrite without sync")
+	}
+
+	conn, err := setupClient()
+	if err != nil {
+		return err
+	}
+	defer func() {
+		err = conn.Close()
+		if err != nil {
+			fmt.Print(err)
+		}
+	}()
+
+	candleInterval := time.Duration(candleGranularity) * time.Second
+
+	s, err := time.Parse(common.SimpleTimeFormat, startTime)
+	if err != nil {
+		return fmt.Errorf("invalid time format for start: %v", err)
+	}
+	e, err := time.Parse(common.SimpleTimeFormat, endTime)
+	if err != nil {
+		return fmt.Errorf("invalid time format for end: %v", err)
+	}
+
+	if e.Before(s) {
+		return errors.New("start cannot be after end")
+	}
```

Comment:
If we move this check up before setupClient() on 708 it will save attempting to open a connection to the grpc server if timestamp parsing fails